File: golang-code/ex14/configuracao.go
package main

import (
	"sync"
)

type Configuracao struct {
	URL string
}

var instancia *Configuracao
var once sync.Once

func GetInstancia() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{}
	})
	
	return instancia
}

File: java-code/src/main/java/ex14/Configuracao.java
package ex14;

public class Configuracao {
    private static Configuracao instancia;
    private String url;

    private Configuracao() {

    }

    public static Configuracao getInstancia() {
        if (instancia == null) {
            instancia = new Configuracao();
        }
        return instancia;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getUrl() {
        return url;
    }
}